scipy/special/special/amos/amos.h

```diff
@@ -5738,6 +5744,16 @@ int amos_unk2(
             kdflg = 2;
             continue;
         }
+        /* GO TO 70 */
+        if (rs1 > 0.0) { return -1; }
+        /* FOR X < 0.0, THE I FUNCTION TO BE ADDED WILL OVERFLOW */
+        if (x < 0.0) { return -1; }
+        kdflg = 1;
+        y[i-1] = 0.0;
+        cs *= -std::complex<double>(0, 1);
+        nz += 1;
+        if (i != 1) { if (y[i-2] != 0.0) { y[i-2] = 0.0;nz += 1; } }
+        continue;
```

Comment:
This fixes a bug from the AMOS Fortran->C translation, 

https://github.com/scipy/scipy/blob/f133c27843a36c8fe1c2979b3a072ac4f5a00624/scipy/special/amos/zunk2.f#L131

the condition above should cause a jump to 70, but this doesn't occur in the logic above without adding these lines here. This seems to have been covered up in the tests by another bug that's been fixed here.